scripts/build.sh

```diff
@@ -16,37 +16,67 @@ GIT_COMMIT=$(git rev-parse HEAD)
 GIT_DIRTY=$(test -n "`git status --porcelain`" && echo "+CHANGES" || true)
 GIT_DESCRIBE=$(git describe --tags)
 
-# If we're building on Windows, specify an extension
-EXTENSION=""
-if [ "$(go env GOOS)" = "windows" ]; then
-    EXTENSION=".exe"
-fi
+# Determine the arch/os combos we're building for
+XC_ARCH=${XC_ARCH:-"386 amd64 arm"}
+XC_OS=${XC_OS:-"darwin freebsd linux netbsd openbsd windows"}
 
-GOPATHSINGLE=${GOPATH%%:*}
-if [ "$(go env GOOS)" = "windows" ]; then
-    GOPATHSINGLE=${GOPATH%%;*}
-fi
+# Install dependencies
+echo "==> Getting dependencies..."
+go get ./...
 
-if [ "$(go env GOOS)" = "freebsd" ]; then
-  export CC="clang"
-fi
+# Delete the old dir
+echo "==> Removing old directory..."
+rm -f bin/*
+rm -rf pkg/*
+mkdir -p bin/
 
-# On OSX, we need to use an older target to ensure binaries are
-# compatible with older linkers
-if [ "$(go env GOOS)" = "darwin" ]; then
-    export MACOSX_DEPLOYMENT_TARGET=10.6
+# If its dev mode, only build for ourself
+if [ "${CONSUL_DEV}x" != "x" ]; then
+    XC_OS=$(go env GOOS)
+    XC_ARCH=$(go env GOARCH)
 fi
 
-# Install dependencies
-echo "--> Installing dependencies to speed up builds..."
-go get \
-  -ldflags "${CGO_LDFLAGS}" \
-  ./...
-
 # Build!
-echo "--> Building..."
-go build \
-    -ldflags "${CGO_LDFLAGS} -X main.GitCommit=${GIT_COMMIT}${GIT_DIRTY} -X main.GitDescribe=${GIT_DESCRIBE}" \
-    -v \
-    -o bin/consul${EXTENSION}
-cp bin/consul${EXTENSION} "${GOPATHSINGLE}/bin"
+echo "==> Building..."
+gox \
+    -os="${XC_OS}" \
+    -arch="${XC_ARCH}" \
+    -ldflags "-X main.GitCommit ${GIT_COMMIT}${GIT_DIRTY} -X main.GitDescribe ${GIT_DESCRIBE}" \
+    -output "pkg/{{.OS}}_{{.Arch}}/consul" \
+    .
+
+# Move all the compiled things to the $GOPATH/bin
+GOPATH=${GOPATH:-$(go env GOPATH)}
+case $(uname) in
+    CYGWIN*)
+        GOPATH="$(cygpath $GOPATH)"
+        ;;
+esac
+OLDIFS=$IFS
+IFS=: MAIN_GOPATH=($GOPATH)
```

Comment:
I'm not 100% sure, I copied it from Terraform. It has to do with putting the binary in the proper place on windows under cygwin.